warzone/Player.cpp

```diff
@@ -1 +1,110 @@
 #include "Player.h"
+
+Player::Player() {  //default constructor
+	this->playerTerritories = new vector<Territory*>();
+	this->playerHand = new Cards::Hand();
+	this->playerOrdersList = new OrdersList();
+}
+
+
+Player::Player(vector<Territory*>* territoriesToAdd, OrdersList* playerList, Cards::Deck* deckToTakeFrom) { //Constructor with all parameters
+	this->playerTerritories = territoriesToAdd;
+	this->playerOrdersList = playerList;
+	this->playerHand = new Cards::Hand(deckToTakeFrom);
+}
+
+Player::~Player() {  //destructor 
+	for (Territory* t : *playerTerritories) {
+		delete t;
+	}
+	delete playerTerritories;
+	delete playerHand;
+	delete playerOrdersList;
+}
+
+Player& Player::operator=(const Player& one)  //assignment operator
+{
+	delete playerHand;
+	this->playerHand = one.playerHand;
+	delete playerTerritories;
+	this->playerTerritories = one.playerTerritories;
+	delete playerOrdersList;
+	this->playerOrdersList = one.playerOrdersList;
+	return *this;
+}
+
+Player::Player(const Player& player)  //copy constructor
+{
+	for (Territory* t : *player.playerTerritories)
+	{
+		Territory temp = new Territory(t);
+		this->playerTerritories->push_back(t);    //copies each territory in the incoming player's territories 
+	}
+	this->playerOrdersList = new OrdersList(*player.playerOrdersList);
+	this->playerHand = new Cards::Hand(*player.playerHand);
+}
+
+const vector<Territory*>* Player::toDefend()   //returns territories the player can defend
+{
+	vector<Territory*>* toDefend = new vector<Territory*>();
+	if (toDefend->empty())
+		cout << "Player has nothing to defend" << endl; //checks if the player has no territories to defend. if so, print that they have none
+	return toDefend;
+}
+
+const vector<Territory*>* Player::toAttack()   //returns territories the player can attack
+{
+	vector<Territory*>* toAttack = new vector<Territory*>();
+	if (toAttack->empty())
+		cout << "Player has nothing to attack" << endl; //checks if the player has no territories to attack. if so, print that they have none
+	return toAttack;
+}
+
+void Player::issueOrder(Order* newOrder)  //allows player to issue an order
+{
+	Order* playerOrder = new Order(*newOrder);
+	playerOrdersList->add(playerOrder);
+}
+
+const vector<Territory*>* Player::getTerritories()  //returns all the player's territories 
+{
+	if (playerTerritories->empty())
+		cout << "Player has no territories" << endl;  //checks if the player's territories are empty. if so, print that they have none
```

Comment:
Idk if we need this print statements and the other print statements like it

Reply:
Will get rid of them! Just wanted to output something to signify the player's territories, hand and orders are all non-existent for now